### ISSUE TITLE
NetworkSimplex's nodeFlexibility pushes edges to the border of nodes

### ISSUE THREAD

**Issue by koenyskout** [eclipse/elk#778]
Hi,

As seen in this [example](http://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUDiANYB0BM+AxnDBAHboAiA8gOoByRTEBHAAVMtHEhxM4FAGbAAbtADKcKMBhgEcAKIEwOAM7pVAVxx8Bw0eMnSDq4HBxgs6btoAqndgCUA0gD6SgCSALJCADLaABoWgiLAYhJSFEw4cADu0GRKMEgADiA4AB6plgBiRcUwAEYwIPBYVDhyJiBwbuys2kHBAFraRHE4aEwAjGgA3hjYECZwaADaaAYwAF446GMADNsANGgATNtoALozNaRoAETclmhj12gAvsOjh1MzWHMLy6sbW12B2OZwuV1u90OT1e-EEowAzJ9MN95ksVutNg8gUcTudMJcQDc7nD4dC3kwACxI2aov4YwH7HFnfHg4kjClk2EjJgAVmpKN+6IBWMZILxwAJRPuPLJhH0owmAFoAHzvNAkPTcpWqpiIjXypgfFWjKn67mI42U9Vy81oS18oA), using nodePlacement.networkSimplex.nodeFlexibility.default: NODE_SIZE somehow yields a weird layout where edges are pushed to the  border:
![Screenshot 2021-08-26 at 17 32 29](https://user-images.githubusercontent.com/4698833/130992267-c1339556-eff8-445e-b488-19ea9541a257.png)
Node 1 is positioned very weirdly to the right, for example. Also, the arrows leaving Node 3 leave at the edges of Node 3, while it would be nicer to have them placed at the center of the space where Node 3 and the nodes below overlap.

Is this a consequence of the algorithm, and is it possible to get a nicer layout through setting some other option?

**Comment by uruuru** [comment]
> would be nicer to have them placed at the center of the space

Definitely. However, the underlying optimization process has no notion of balance/symmetry.  

> and is it possible to get a nicer layout through setting some other option?

Unfortunately, no.

**Comment by koenyskout** [comment]
Thanks for the answers, I feared as much.

For now, I'll try to see if layouting in 2 passes could work: first with nodeFlexibility to determine appropriate node sizes, and then another one on the output of that but without the nodeFlexibility option set, in the hope that that puts the edges at a better location.

**Comment by soerendomroes** [comment]
#1194 Fixes this. Another options is adding portsurrounding spacing.
